### PR TITLE
Add eg section

### PR DESCRIPTION
**Pull request by LOZORD**

### CODE REVIEW

File: 1-source-code/2-writing/2-gofmt-and-friends.md [hunk 1]
```diff
@@ -302,6 +302,59 @@ type Person struct {
...
+$ go get golang.org/x/tools/cmd/eg
+$ ^get^build
+```
+
+### Writing a template
+
+Here's a simple template (`template.go`) to get you started.
+Templates need to be buildable Go files with `before` and `after` functions.
+
+```go
```
Comment (campoy) [inline 1-source-code/2-writing/2-gofmt-and-friends.md]: please use embedmd for this, see an example of how it works in https://github.com/campoy/go-tooling-workshop/blob/master/1-source-code/1-workspace/1-intro.md

File: 1-source-code/2-writing/2-gofmt-and-friends.md [hunk 2]
```diff
@@ -302,6 +302,59 @@ type Person struct {
...
+
+`eg` is an example-based refactoring tool.
+To use it, you simply create a template file and then provide the paths you wish to refactor.
+`eg` is particulary useful for changing function or API usages.
+
+### Installing `eg`
+
+```bash
+$ go get golang.org/x/tools/cmd/eg
+$ ^get^build
```
Comment (campoy) [inline 1-source-code/2-writing/2-gofmt-and-friends.md]: what does this mean?

File: 1-source-code/2-writing/2-gofmt-and-friends.md [hunk 3]
```diff
@@ -302,6 +302,59 @@ type Person struct {
...
+	"fmt"
+)
+
+func before(s string) error { return fmt.Errorf("%s", s) }
+func after(s string) error  { return errors.New(s) }
+```
+
+### Running `eg`
+
+```bash
```
Comment (campoy) [inline 1-source-code/2-writing/2-gofmt-and-friends.md]: Separate this into some Go code using embedmd, then explain that running that command (in a bash block) will change the file into something else

File: 1-source-code/2-writing/2-gofmt-and-friends.md (hunk 2 above)
Comment (LOZORD) [reply 1-source-code/2-writing/2-gofmt-and-friends.md]: It is a command substitution hack: https://www.cyberciti.biz/faq/bash-history-repeat-substitution-command-syntax/

Basically, it looks for the first occurrence of `get` and replaces it with `build`.

__However__, I think this is basically the behavior of `go install`, right? I'll replace these two lines with that.

File: 1-source-code/2-writing/2-gofmt-and-friends.md (hunk 1 above)
Comment (LOZORD) [reply 1-source-code/2-writing/2-gofmt-and-friends.md]: Done! This is a great tool :smile_cat:

File: 1-source-code/2-writing/2-gofmt-and-friends.md (hunk 3 above)
Comment (LOZORD) [reply 1-source-code/2-writing/2-gofmt-and-friends.md]: Good idea -- done.

**Comment by LOZORD** [comment]
Sorry for the delay! I've addressed your comments ... finally :sweat_smile:

**Pull request merged**